File: 977. Squares of a Sorted Array/main.py
class Solution:
    def sortedSquares(self, nums: List[int]) -> List[int]:
        k = len(nums) - 1
        result = [0] * k

        i, j = 0, len(nums) - 1
        while i <= j:
            if abs(nums[i]) > abs(nums[j]):
                result[k] = nums[i] ** 2
                i += 1
            else:
                result[k] = nums[j] ** 2
                j -= 1
            k -= 1
        
        return result

File: 977. Squares of a Sorted Array/main.go
func sortedSquares(nums []int) []int {
    res := make([]int, len(nums))
	k := len(res) - 1

	abs := func(a int) int {
		if a < 0 {
			return -a
		}
		return a
	}
	
	i, j := 0, len(nums) - 1
	for i <= j {
		if abs(nums[i]) > abs(nums[j]) {
			res[k] = nums[i]*nums[i]
			i++
		} else {
			res[k] = nums[j]*nums[j]
			j--
		}
		k--
	}

    return res
}